src/analysis/normed_space/inner_product.lean

```diff
@@ -1851,4 +1851,30 @@ begin
   rw add_zero at hd,
   exact hd.symm
 end
+
+/-- Given a finite-dimensional subspace `K₂`, and a subspace `K₁`
+containined in it, the dimensions of `K₁` and the intersection of its
+orthogonal subspace with `K₂` add to that of `K₂`. -/
+lemma submodule.findim_add_inf_findim_orthogonal' {K₁ K₂ : submodule 𝕜 E}
+  [finite_dimensional 𝕜 K₂] (h : K₁ ≤ K₂) {n : ℕ} (h_dim : findim 𝕜 K₁ + n = findim 𝕜 K₂) :
+  findim 𝕜 (K₁ᗮ ⊓ K₂ : submodule 𝕜 E) = n :=
+by { rw ← add_right_inj (findim 𝕜 K₁), simp [submodule.findim_add_inf_findim_orthogonal h, h_dim] }
+
+/-- Given a finite-dimensional space `E` and subspace `K`, the dimensions of `K` and `Kᗮ` add to
+that of `E`. -/
+lemma submodule.findim_add_findim_orthogonal [finite_dimensional 𝕜 E] {K : submodule 𝕜 E} :
+  findim 𝕜 K + findim 𝕜 Kᗮ = findim 𝕜 E :=
+begin
+  have : findim 𝕜 E = findim 𝕜 (⊤ : submodule 𝕜 E) := findim_top.symm,
+  have : Kᗮ = Kᗮ ⊓ ⊤ := inf_top_eq.symm,
+  convert submodule.findim_add_inf_findim_orthogonal (le_top : K ≤ ⊤)
```

Comment:
This proof smell like something is wrong in the API. Are there missing lemmas?

Reply:
I pushed a variant proof, see if you like it better.  While playing with this I concluded that what makes it awkward is the need for the fact `findim 𝕜 ↥Kᗮ = findim 𝕜 ↥(Kᗮ ⊓ ⊤)`, or more generally, the following fact, which doesn't work as a simp-lemma:
```
example {A B : submodule 𝕜 E} (h : A = B) : findim 𝕜 A = findim 𝕜 B := by rw h
```
Ideas welcome!